### PR TITLE
Setting up ground truth masks for image segmentation

### PR DESCRIPTION
**Pull request by romerocruzsa**

### CODE REVIEW

**Comment by romerocruzsa** [comment]
## Summary of NCNet Model Performance

1. **Training Details:**
   - **Loss Function:** Binary Cross Entropy (BCEWithLogitsLoss)
   - **Optimizer:** Adam
   - **Learning Rate:** 1e-3

   image

2. **Testing/Inference Performance Metrics (k=1):**
   - **BCELoss:** 0.235782
   - **Accuracy:** 0.897461
   - **Precision:** 0.940296
   - **Recall:** 0.854526
   - **F1 Score:** 0.891905
   - **AUC Score:** 0.966644
   - **Confusion Matrix:** `[[475, 25], [77, 423]]`

   image

   image

3. **Testing/Inference Performance Metrics (k=100), Best 10 Accuracies:**

   | k   | BCELoss | Accuracy | Precision | Recall | F1 Score | AUC Score | Confusion Matrix          |
   |-----|---------|----------|-----------|--------|----------|-----------|---------------------------|
   | 45  | 0.253583| 0.889648 | 0.930123  | 0.830398 | 0.874869 | 0.956703  | `[[473, 27], [77, 423]]`  |
   | 6   | 0.238760| 0.889648 | 0.945617  | 0.820253 | 0.874283 | 0.964692  | `[[476, 24], [86, 414]]`  |
   | 69  | 0.245434| 0.889648 | 0.940076  | 0.837775 | 0.883624 | 0.959457  | `[[475, 25], [79, 421]]`  |
   | 7   | 0.243916| 0.890625 | 0.943705  | 0.832273 | 0.882074 | 0.963432  | `[[477, 23], [80, 420]]`  |
   | 39  | 0.242993| 0.890625 | 0.945942  | 0.836871 | 0.883847 | 0.963791  | `[[475, 25], [81, 419]]`  |
   | 64  | 0.246401| 0.890625 | 0.949697  | 0.829912 | 0.876933 | 0.967866  | `[[477, 23], [80, 420]]`  |
   | 43  | 0.248743| 0.891602 | 0.940297  | 0.830182 | 0.879750 | 0.958842  | `[[478, 22], [80, 420]]`  |
   | 40  | 0.239675| 0.891602 | 0.951502  | 0.832718 | 0.879977 | 0.966831  | `[[477, 23], [82, 418]]`  |
   | 79  | 0.244398| 0.891602 | 0.931745  | 0.837029 | 0.879347 | 0.961393  | `[[473, 27], [78, 422]]`  |
   | 37  | 0.250139| 0.891602 | 0.938748  | 0.847000 | 0.887661 | 0.965443  | `[[473, 27], [78, 422]]`  |
